src/main/java/com/redhat/jenkins/plugins/bayesian/Utils.java

```diff
@@ -45,10 +44,23 @@
         // Maven
         List<FilePath> mavenManifests = findManifestsFromList(workspace, knownMavenManifests);
         if (!mavenManifests.isEmpty()) {
-            // TODO: get all poms
-            FilePath pom = workspace.child("target/stackinfo/poms/pom.xml");
-            if (manifestExists(pom)) {
-                manifests.add(pom);
+            List<FilePath> childPomDirs = new ArrayList<FilePath>();
+            FilePath rootPom = workspace.child("target/stackinfo/poms/pom.xml");
+            if (manifestExists(rootPom)) {
+                manifests.add(rootPom);
+            }
+
+            FilePath childPomDirsPath = workspace.child("target/stackinfo/poms");
+            try {
+                childPomDirs = childPomDirsPath.listDirectories();
```

Comment:
Note there can be pom.xml files nested even deeper, i.e. you can have `target/stackinfo/poms/module-1/another-module/pom.xml`. This will only cover the first level, if `listDirectories()` method does what I think it does.